torch/testing/_comparison.py

```diff
@@ -776,11 +776,21 @@ def _compare_sparse_csr_values(
     ) -> None:
         """Compares sparse CSR tensors by comparing
 
+        - the shape
         - the number of non-zero elements (nnz) for equality,
         - the col_indices for equality,
         - the crow_indices for equality, and
         - the values for closeness.
         """
+        def raise_mismatch_error(attribute_name: str, actual_value: Any, expected_value: Any) -> NoReturn:
+            raise self._make_error_meta(
+                AssertionError,
+                f"The values for attribute '{attribute_name}' do not match: {actual_value} != {expected_value}.",
+            )
+
+        if actual.shape != expected.shape:
+            raise_mismatch_error("shape", actual.shape, expected.shape)
```

Comment:
@cpuhrsch This seems redundant. The comparison is dispatched from the `compare` method. In there we call `_compare_attributes` before we get to the value comparison in `_compare_sparse_csr_values`:

https://github.com/pytorch/pytorch/blob/c755d220450a8d132dff6d8a56711d93a2d17abd/torch/testing/_comparison.py#L611-L616 

`_compare_attributes` performs the exact shape you added here:

https://github.com/pytorch/pytorch/blob/c755d220450a8d132dff6d8a56711d93a2d17abd/torch/testing/_comparison.py#L641-L648

Was there a case where this was not triggered?

Reply:
I added this to make sure it won't be missed based on the case for the strided comparison. Happy to remove it again now that you're pointing this out.